### PR TITLE
Add content type to attachments

### PR DESCRIPTION
**Pull request by JayDwee**
New to discord.py 1.7 is 'content_type' for attachments

This adds an optional param to the make_attachment_dict factory

### CODE REVIEW

File: discord/ext/test/factories.py [hunk 1]
```diff
@@ -293,17 +293,18 @@ def dict_from_message(message: discord.Message):
     return out
 
 
-def make_attachment_dict(filename, size, url, proxy_url, id_num=-1, height=None, width=None):
+def make_attachment_dict(filename, size, url, proxy_url, id_num=-1, height=None, width=None, content_type=None):
     if id_num < 0:
         id_num = make_id()
     return {
         'id': id_num,
-        'filename': filename,
```
Comment (Sergeileduc) [inline discord/ext/test/factories.py:301]: Is there a reason why `filename` is after stuffs like `size, height, or width` ?

Seems odd to me...

**Review by Sergeileduc — CHANGES_REQUESTED** [review]
Hi !

Seems fine (I trust you with the `guild_needs_chuniking` stuff, and attachment content type.

But why change the order in `make_attachment_dict` ?

first items should be `id` and `filename`, as it was.

**Comment by JayDwee** [comment]
> Hi !
> 
> Seems fine (I trust you with the `guild_needs_chuniking` stuff, and attachment content type.)
> 
> But why change the order in `make_attachment_dict` ?
> 
> first items should be `id` and `filename`, as it was.

I changed it to the order used in discord py documentation. In the end it doesn't effect anything

**Comment by Sergeileduc** [comment]
> I changed it to the order used in discord py documentation. In the end it doesn't effect anything

Still, it's weird (and add a diff (a git diff) that is not necessary), and no matter what the documentation says, the function takes the filename as 1st parameter, so...

**Comment by JayDwee** [comment]
Fixed @Sergeileduc

**Review by Sergeileduc — APPROVED** [review]

**Pull request merged**